sympy/combinatorics/coset_table.py

```diff
@@ -783,7 +810,7 @@ def modified_define(self, alpha, x):
             raise ValueError("the coset enumeration has defined more than %s cosets. Try with a greater value max number of cosets "%C.coset_table_max_limit)
         self.table.append([None]*len(self.A))
         H = self.subgroup
-        self._grp = (list(free_group(', ' .join(["a_%d" % i for i in range(len(H))])))).pop(0)
+        self._grp = list(free_group(', ' .join(["a_%d" % i for i in range(len(H))])))[0]
```

Comment:
This can just be `free_group(', ' .join(["a_%d" % i for i in range(len(H))])))[0]` - no need for `list`.